src/plugin_output_processing/provider.py

```diff
@@ -18,60 +18,77 @@
 
 import ollama
 import openai
+from httpx import ConnectError
 from loguru import logger
 
+OLLAMA_NAME = "ollama"
+OPENAI_NAME = "openai"
+
 
 class ProviderError(Exception):
     pass
 
 
 class Provider:
-
     name: str
-    models: list[str] = []
-    available: bool
+    available: bool = False
     url: str = None
-
-    def get_model(self, model) -> str:
-        if not model:
-            logger.warning(f"Model not set for {self.name}. Using default.")
-            model = self.default
-        if model not in self.models:
-            logger.warning(f"Model {model} not found in {self.name} models.")
-            model = self.default
+    model: str
+
+    def get_model(
+        self,
+        list_models: list[str],
+        model: str | None = None,
+        default_model: str | None = None,
+    ) -> str:
+        if default_model is None:
+            default_model = list_models[0]
+        if model is None:
+            return default_model
+        if model not in list_models:
+            logger.warning(f"{model} not found in model list, using default instead.")
+            return default_model
         return model
 
 
 class Ollama(Provider):
-
-    name = "ollama"
+    name = OLLAMA_NAME
     url = f"http://{os.environ.get('OLLAMA_HOST', 'localhost')}:11434"
 
-    def __init__(self) -> None:
-        self.models = self._list_models()
-        self.default = self.models[0] if self.models else None
-        self.available = self.models != []
+    def __init__(self, model: str = None) -> None:
+        self.model = self.fetch_model(model)
+        self.available = self.model is not None
```

Comment:
This constructor is the same for the `OpenAI` class. Maybe, we can move it in the base class `Provider`

Reply:
Agree, thanks!

Reply:
I think we don't need to define an `__init__` method for the `Ollama` class because it inherites the one from `Provider`.

Reply:
My bad, you were right